test/extensions/filters/http/grpc_json_transcoder/grpc_json_transcoder_integration_test.cc

```diff
@@ -264,6 +264,22 @@ TEST_P(GrpcJsonTranscoderIntegrationTest, TestParamUnescapePlus) {
       R"({"id":"20","theme":"Children"})");
 }
 
+TEST_P(GrpcJsonTranscoderIntegrationTest, QueryParams111) {
+  HttpIntegrationTest::initialize();
+  testTranscoding<bookstore::CreateShelfRequest, bookstore::Shelf>(
+      Http::TestRequestHeaderMapImpl{{":method", "POST"},
+                                     {":path", "/shelf?shelf.search[field]=Google"},
```

Comment:
As @lizan mentions in #21151 [here](https://github.com/envoyproxy/envoy/issues/21151#issuecomment-1145216299) `[` and `]` aren't valid characters in query params so they will need to be escaped (many browsers will do this for you automatically, so even if you type `[` and `]` in the query param the browser actually sends  `%5B` and `%5D`)

This means that the path envoy gets is actually "/shelf?shelf.search%5Bfield%5D=Google"

Does the test still pass if you use that as the path?

Reply:
[Here is the RFC](https://datatracker.ietf.org/doc/html/rfc3986#section-2.2) and you will see that `[` and `]` are reserved characters, and if you want to use them in the query param you need to [percent encode them](https://datatracker.ietf.org/doc/html/rfc3986#section-2.1).

Reply:
Fixed it